### PR TITLE
Network pre-checks as a new patch

### PR DESCRIPTION
**Pull request by divya-vijay**
Splitting the pre-checks as a separate patch for easy review:
1. Has Network mellanox driver check
2. Network mellanox HCA presence and ports check

### CODE REVIEW

File: py-utils/src/utils/validator/v_network.py [hunk 1]
```diff
@@ -31,21 +31,47 @@ def validate(self, v_type, args):
...
-                errno.EINVAL, "Action parameter %s not supported" % v_type)
+            if len(args) < 2:
+                raise VError(errno.EINVAL,
+                             "Insufficient parameters. %s" % args)
+
+            else:
+                if v_type == "passwordless":
+                    self.validate_passwordless_ssh(args[0], args[1:])
+                elif v_type == "drivers":
+                    nodes = ['srvnode-1', 'srvnode-2']
```
Comment (ujjwalpl) [inline py-utils/src/utils/validator/v_network.py:57]: nodes should be passed as parameters. No hard coding in the code

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/validator/v_network.py [hunk 2]
```diff
@@ -106,3 +132,52 @@ def _resolve_host(self, host):
...
+                       f"Also, check if '{node}' is valid. "
+                       f"CMD {cmd} failed. {run_result[0]}. {run_result[1]}")
+                raise VError(errno.EINVAL, res)
+
+    def validate_hca(self, provider, nodes):
+        """Check if HCA presence and ports"""
+
+        for node in nodes:
+            if provider.lower() == "mellanox":
+                cmd = f"ssh {node} lspci -nn | grep 'Mellanox Technologies'"
```
Comment (ujjwalpl) [inline py-utils/src/utils/validator/v_network.py:155]: Mellanox can be passed as input and not hardcoded.

File: py-utils/src/utils/validator/v_network.py [hunk 3]
```diff
@@ -106,3 +132,52 @@ def _resolve_host(self, host):
...
+                if ports_run_result[1] or ports_run_result[2]:
+                    res = ("Mellanox Host Channel Adapters ports were not "
+                           "detected by command 'lspci -nn | grep 'Mellanox Technologies'. "
+                           f"Also, check if '{node}' is valid. "
+                           f"CMD '{cmd} failed. {ports_run_result[0]}. {ports_run_result[1]}")
+                    raise VError(errno.EINVAL, res)
+
+                res = (ports_run_result[0].decode('utf-8').strip())
+                if int(res) == 0:
+                    res = ("Mellanox Host Channel Adapters ports were not "
```
Comment (ujjwalpl) [inline py-utils/src/utils/validator/v_network.py:179]: Avoid hard coding

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/validator/v_network.py (hunk 1 above)
Comment (divya-vijay) [reply py-utils/src/utils/validator/v_network.py]: @ujjwalpl nodes were not hardcoded for passing as arg, this was to ensure driver name is not empty.
Comment addressed and code updated now.

File: py-utils/src/utils/validator/v_network.py (hunk 3 above)
Comment (divya-vijay) [reply py-utils/src/utils/validator/v_network.py]: Yes, code updated to get the provider value dynamically. It looks like, 
network hca <provider> nodes

File: py-utils/src/utils/validator/v_network.py (hunk 2 above)
Comment (divya-vijay) [reply py-utils/src/utils/validator/v_network.py]: Sure, Thanks. 
Code is now updated to get the provider value dynamically.

**Review by ujjwalpl — APPROVED** [review]

**Review by ajaysrivas — APPROVED** [review]

**Pull request merged**